services/tests/test_models.py

```diff
@@ -55,41 +59,86 @@ def test_allowed_data_fields_get_correct_orders(allowed_data_field_factory):
     assert second_data_field.order == 2
 
 
+@pytest.mark.parametrize("service__gdpr_url", [GDPR_URL])
 def test_download_gdpr_data_with_valid_service_and_url(
-    requests_mock, youth_profile_response, service_factory, profile
+    requests_mock, youth_profile_response, service, profile
 ):
-    # setup models
-    service = service_factory(gdpr_url="http://valid-gdpr-url.com/profiles/")
     service_connection = ServiceConnectionFactory(profile=profile, service=service)
-    # mock request
     requests_mock.get(
-        f"http://valid-gdpr-url.com/profiles/{profile.pk}", json=youth_profile_response
+        f"{GDPR_URL}{profile.pk}",
+        json=youth_profile_response,
+        request_headers={"authorization": "Bearer token"},
     )
 
-    response = service_connection.download_gdpr_data()
-    assert response.json() == youth_profile_response
+    response = service_connection.download_gdpr_data(api_token="token")
+    assert response == youth_profile_response
 
 
-def test_download_gdpr_data_with_invalid_service(
-    requests_mock, service_factory, profile, service
-):
-    # setup models
+def test_download_gdpr_data_with_invalid_service(requests_mock, profile, service):
```

Comment:
This test name should state how the service is invalid. I can't tell that from the test body.